plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,176 @@
+    // `48 - RANDOM_SUFFIX_LENGTH` for the exact value.
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
+    {
+        this.sqlExecutor = sqlExecutor;
+        this.keyspace = keyspace;
+        this.tableName = namePrefix + randomTableSuffix();
+        sqlExecutor.execute(format("CREATE TABLE %s.%s %s", keyspace, tableName, tableDefinition(columnDefinitions)));
+        try {
+            String columns = columnDefinitions.stream()
+                    .map(columnDefinition -> columnDefinition.name)
+                    .collect(joining(", "));
+            for (String row : rowsToInsert) {
+                // some databases do not support multi value insert statement
+                sqlExecutor.execute(format("INSERT INTO %s.%s (%s) VALUES (%s)", keyspace, tableName, columns, row));
+            }
+
+            // Ensure that the currently created table is visible to other sessions (e.g. the session used in DistributedQueryRunner)
+            server.refreshSizeEstimates(keyspace, tableName);
+        }
+        catch (Exception e) {
+            try (TestCassandraTable ignored = this) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    public String getTableNameInKeyspace()
+    {
+        return format("%s.%s", keyspace, tableName);
+    }
+
+    private static String tableDefinition(List<ColumnDefinition> columnDefinitions)
+    {
+        ImmutableList.Builder<String> partitionColumns = ImmutableList.builder();
+        ImmutableList.Builder<String> clusterColumns = ImmutableList.builder();
+        for (ColumnDefinition columnDefinition : columnDefinitions) {
+            if (columnDefinition.primaryKeyType == PrimaryKeyType.PARTITION) {
+                partitionColumns.add(columnDefinition.name);
+            }
+            else if (columnDefinition.primaryKeyType == PrimaryKeyType.CLUSTER) {
+                clusterColumns.add(columnDefinition.name);
+            }
+        }
+        List<String> partitions = partitionColumns.build();
+        List<String> clusters = clusterColumns.build();
+
+        verify(partitions.size() > 0, "Cassandra table must have at least one partition key");
+
+        String primaryKey = format(
+                "(%s%s)",
+                partitions.stream().collect(joining(", ", "(", ")")),
+                clusters.isEmpty() ? "" : clusters.stream().collect(joining(", ", ", ", "")));
+
+        return columnDefinitions.stream()
+                .map(columnDefinition -> columnDefinition.name + " " + columnDefinition.type)
+                .collect(joining(",\n", "(\n", ",\nPRIMARY KEY " + primaryKey + "\n)"));
```

Comment:
nit: No need to format SQL here because the generated query is not printed. 
```suggestion
                .collect(joining(",", "(", ",PRIMARY KEY " + primaryKey + ")"));
```